collection+xattrs.go

```diff
@@ -618,16 +619,74 @@ func removeUserXattrs(xattrs semiParsedXattrs) {
 	}
 }
 
+// macroExpand takes the macro and assigned 'value' the correct value based on what the macro is provided (this is gocb macro format)
+func (e *event) macroExpand(value string, casServerFormat string, bucketUUID string) string {
+	if value == "\"${Mutation.CAS}\"" {
+		value = casServerFormat
+	} else if value == "\"${Mutation.value_crc32c}\"" {
+		value = encodedCRC32c(e.value)
+	} else {
+		// if the value isn't to be macro expanded to CAS or CRC hash then it is a source ID
+		value = bucketUUID
+	}
+	return value
+}
+
 // Sets JSON properties "cas" to the given `cas`, and "value_crc" to CRC checksum of `docValue`.
-func (e *event) expandSyncXattrMacros(xattr any) {
+func (e *event) expandSyncXattrMacros(xattr any, bucketUUID string, mutateSpec *sgbucket.MutateInOptions) error {
+	var err error
+	if mutateSpec == nil {
+		emptySpec := sgbucket.MutateInOptions{}
```

Comment:
If `mutateSpec` is nil this function won't do anything, so it might as well just return here.